### ISSUE TITLE
bug: "Open-With" opens the app, but not the video. 

### ISSUE THREAD

**Issue by Abhinav1217** [revanced/revanced-patches#788]
### Type

Error at runtime

### Bug description

When, from a google search, or some website, I click on "open in youtube",  If I choose revanced as my app, it simply opens the app and does not go to video. If I choose any other valid apps, it opens the video in the apps.

I don't know how to record my phone screen so here is just a screenshot of my app chooser menu when I click on a youtube video from  a google search result.
  
![Screenshot_20221014-141244](https://user-images.githubusercontent.com/740584/195805326-2b65aada-5100-49b7-b4df-c4d613f79470.jpg)

Here if I choose revanced, it will open the app, but not go to the video, But if I choose newpipe, or vuetube or vanced, it will play video on the app. 

### Steps to reproduce

1 Search some video on google search app or browser
2 Click on "open with" options, to show selection of external apps that can handle youtube links

3 Click on revanced 
- Expected: Open the video on app.
- Actual: Only opened the app. 

3 Clicked on alternatives like newpipe or vanced
- Open the app and played the video. 

### Relevant log output

```shell
Where to get it?
```

### Additional context

In the screenshot, you are seeing icon for revanced extended, But I have tried it on revanced too. Because it was not working on revanced, someone suggested trying revanced extended builds and I installed that too. But it was same issue on rvx too so I took a screenshot and am submitting the bug report.

### Acknowledgements

- [X] I have searched the existing issues and this is a new and no duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I filled out all of the requested information in this issue properly.

**Comment by oSumAtrIX** [comment]
Disable patches until the error is gone, report back which one caused it.

**Comment by SodaWithoutSparkles** [comment]
I have this issue before. The issue also happened in revanced. 

Solution: tap back, click the link again, select revanced again, then you got the video

Please try it out, and if it works, I think this issue can be closed

**Comment by Abhinav1217** [comment]
@SodaWithoutSparkles 
> Solution: tap back, click the link again, select revanced again, then you got the video

No, Not the solution. I won't bother the devs if issue was this mundane.  I have already done troubleshooting basics before submitting issues here, including rebuilding with different builders. 


> Disable patches until the error is gone, report back which one caused it.

I am having troubles on this. I disabled all the patches, then enabled just one, The issue came back. Then I build another one by disabling all patches and enabling another one, the issue came back. It feels like the patching is the issue not the patches. But I can't test a build without any patches enabled because that won't install because of some signature error, so I can't be sure if this is because of the yt apk the script is downloading from apkmirror or some other reason. 

I am using both [revanced-builder](https://github.com/reisxd/revanced-builder) and [rvx-builder](https://github.com/inotia00/rvx-builder) equally, on my linux laptop to build my apks.

**Comment by oSumAtrIX** [comment]
You need to set defaults in the settings of the app in the preferences your Android phone.

**Comment by Abhinav1217** [comment]
> You need to set defaults in the settings of the app

If I did that, I wont be able to choose between vanced, revanced or some other app for youtube link.  Currently The issue only exists on revanced app, not vanced, or newpipe.

> Also please make a recording of how the error looks like, I can see Vanced there. 

Any tips on how do I do that? My phone does have a screen recorder feature but it is only enabled inside some games as part of game mode. Should I try scrcpy to record on my laptop?

**Comment by SodaWithoutSparkles** [comment]
I guess its MIUI? then there should be an built-in app to do so. Just swipe down from top right corner and press edit icons layout, then look for an recorder icon. If theres a green plus at top right of the icon, click it ti add to the quick-access menu

**Comment by Abhinav1217** [comment]
There is this setting in Revanced Extended, but not in Revanced. 

![Screenshot_20221110-153830](https://user-images.githubusercontent.com/740584/201063374-7fa879bd-59a3-43b9-9054-a86dce322a55.jpg)

This seems to fix the issue. I tried building a normal revanced apk, using reisxd/revanced-builder but I couldn't find this setting there. 

Sorry it took so much time to reply, I could only try few different builds each day, and a lot of time, the app just crashed quickly. and this setting was disabled by default, so even with rvx-builder, I think I missed this. I only caught this today because I was helping it installed and configured on a friend's device. 

If this patch is available in revanced, that I missed, please close this issue.  

Thanks for all your help and suggestions.

**Comment by Abhinav1217** [comment]
I don't know, What was the issue then. On your suggestion, I tried enabling plugins one by one, but basically for any patched version, "Open with" revanced, didn't load the video. And I made sure every patch was updated, by deleting the cache and download folder before every build.  I was in patch number 9 from top. 

This was something I just happen to get by chance, I built a revanced extended apk for a friend with every patch included, and on his phone, It was working. I tried it on my own phone, and behold, now opening yt links with revanced, opened the video. 

> The option exists in ReVanced natively

This option resides in "Settings->Revanced Settings -> Extended settings". I will try rebuilding not-extended revanced apk, to see if this is something that was solved in rvx,

> it does not change anything related to opening links with the app. 
Apperently, you are right, I disable this setting and was still able to open videos normally. 


Is there a way to get to the root of the issue by decompiling apk, I still have the original apk on which I reported the issue.
